batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/BatchJobExecutionResource.java

```diff
@@ -216,9 +229,13 @@ private boolean hasJobExecution(final String executionName) {
      * guarded.
      */
     private void refreshChildren() {
+        if (System.currentTimeMillis() - lastRefreshedTime < refreshMinInterval) {
+            return;
+        }
+
         final List<JobExecution> executions = new ArrayList<>();
         // Casting to (Supplier<List<JobInstance>>) is done here on purpose as a workaround for a bug in 1.8.0_45
-        final List<JobInstance> instances = jobOperator.allowMissingJob((Supplier<List<JobInstance>>)() -> jobOperator.getJobInstances(jobName, 0, jobOperator.getJobInstanceCount(jobName))
+        final List<JobInstance> instances = jobOperator.allowMissingJob((Supplier<List<JobInstance>>)() -> jobOperator.getJobInstances(jobName, 0, 100)
```

Comment:
This actually changes the current behavior by limiting a maximum of 100 job. I'm not sure we should be changing this. At a minimum we should be making it configurable somehow with it defaulting to the current behavior.

Reply:
Yeah, I was't sure about this either. I think we can preserve the current behavior (I'll change the limit to be Integer.MAX, which will down match the actual count). Hope by avoiding excessive refresh, this can mitigate the long deployment problem.

In the long term, we should optimize this part to use a single operation (a query in jdbc job repository), which will need an upgrade of jberet-core. So the limit here will no longer be needed.